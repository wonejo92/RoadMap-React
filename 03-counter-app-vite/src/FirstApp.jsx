import PropTypes from "prop-types";

export const FirstApp = ({title,subTitle,numbers}) => {
  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subTitle}</p>
    <p>{subTitle}</p>
    <p>{numbers+10}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  numbers: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  title:"Default Title",
  subTitle:"PropTypes",
  numbers:22,
  name: "KEVIN CORDERO"
}