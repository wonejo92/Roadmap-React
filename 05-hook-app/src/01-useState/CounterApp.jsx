import {useState} from "react";

export const CounterApp = () => {
    const [state, setCounter] = useState({
        counter1:10,
        counter2:20,
        counter3:30
    });

    const{counter1,counter2,counter3} = state;

    return(
      <>
          <h1>COUNTER: {counter1}</h1>
          <h1>COUNTER: {counter2}</h1>
          <h1>COUNTER: {counter3}</h1>
          <hr/>
          <button className="btn" onClick={()=>setCounter({
              ...state, //Permite Modificar en este caso solo counter1 y sumarle +1 y el resto de counter se mantienem. 
              counter1:counter1+1,
          })}>+1</button>
      </>
  )
}