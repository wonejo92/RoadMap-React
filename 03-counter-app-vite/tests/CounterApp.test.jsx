import {fireEvent, render, screen} from '@testing-library/react';
import {FirstApp} from "../src/FirstApp.jsx";
import {CounterApp} from "../src/CounterApp.jsx";
describe('CounterApp Test', () => {

    const value=10

    test('must match with the snapshot', () => {
        const{container} = render(<CounterApp value={value} />);
        // screen.debug()
        expect(container).toMatchSnapshot();
    })

    test('must show 100 ', () => {
        render(<CounterApp value={100}  />);
        expect(screen.getByText('100')).toBeTruthy()
        // expect(screen.getByRole('heading', { level:2 })).toBeTruthy()
    })

    // SIMULACION DE EVENTOS (CLICK)
    test('one click simulation to increment with +1 button', () => {
        render(<CounterApp value={value}  />);
        fireEvent.click(screen.getByText('+1'))
        expect(screen.getByText('11')).toBeTruthy()
    })

    test('one click simulation to dicrement with -1 button', () => {
        render(<CounterApp value={value}  />);
        fireEvent.click(screen.getByText('-1'))
        expect(screen.getByText('9')).toBeTruthy()
    })

    test('one click simulation to reset with reset button', () => {
        render(<CounterApp value={value}  />);
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('Reset'))
        expect(screen.getByText('10')).toBeTruthy()
    })
})