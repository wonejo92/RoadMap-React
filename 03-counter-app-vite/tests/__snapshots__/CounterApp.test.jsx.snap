// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp Test must match with the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
